`init` command fails when executing with `bunx`

### What happened?

Currently when executing:
```bash
bunx react-native-test-app@latest init
```

it fails when en error:

```
Error: Cannot find module 'react-native/package.json'
Require stack:
- /private/tmp/bunx-501-react-native-test-app@latest/node_modules/react-native-test-app/scripts/helpers.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Function.resolve (node:internal/modules/helpers:190:19)
    at getPackageVersion (/private/tmp/bunx-501-react-native-test-app@latest/node_modules/react-native-test-app/scripts/helpers.js:144:32)
    at file:///private/tmp/bunx-501-react-native-test-app@latest/node_modules/react-native-test-app/scripts/configure.mjs:740:29
    at parseArgs (file:///private/tmp/bunx-501-react-native-test-app@latest/node_modules/react-native-test-app/scripts/utils/parseargs.mjs:103:5)
    at file:///private/tmp/bunx-501-react-native-test-app@latest/node_modules/react-native-test-app/scripts/configure.mjs:705:3
    at ModuleJob.run (node:internal/modules/esm/module_job:234:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:473:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:123:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/private/tmp/bunx-501-react-native-test-app@latest/node_modules/react-native-test-app/scripts/helpers.js'
  ]
}
```

### Version

0.4.3

### What platforms are you seeing this issue on?

- [ ] Android
- [ ] iOS
- [ ] macOS
- [ ] visionOS
- [ ] Windows

### System Information

```true
System:
  OS: macOS 15.3.1
  CPU: (8) arm64 Apple M1 Pro
  Memory: 579.56 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.17.0
    path: ~/.local/state/fnm_multishells/1926_1742286632334/bin/node
  Yarn:
    version: 1.22.22
    path: ~/.local/state/fnm_multishells/1926_1742286632334/bin/yarn
  npm:
    version: 10.8.2
    path: ~/.local/state/fnm_multishells/1926_1742286632334/bin/npm
  Watchman:
    version: 2025.03.03.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.16.2
    path: /Users/szymonrybczak/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK:
    Android NDK: 22.1.7171670
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12700392
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 20.0.2
    path: /usr/bin/javac
  Ruby:
    version: 2.7.5
    path: /Users/szymonrybczak/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.1
    wanted: 15.0.1
  react:
    installed: 19.0.0
    wanted: 19.0.0
  react-native:
    installed: 0.78.0
    wanted: 0.78.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```

### Steps to Reproduce

1. Run `bunx react-native-test-app@latest init` 

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.yarn/plugins/boost-workaround.cjs
+++ b/.yarn/plugins/boost-workaround.cjs
@@ -5,11 +5,11 @@
  * @typedef {{ configuration: Configuration; cwd: string; workspaces: Workspace[]; }} Project
  * @typedef {{ mode?: "skip-build" | "update-lockfile"; }} InstallOptions
  *
- * @type {{ name: string; factory: (require: NodeRequire) => unknown; }}
+ * @type {{ name: string; factory: (require: NodeJS.Require) => unknown; }}
  */
 module.exports = {
   name: "plugin-boost-workaround",
-  factory: (_require) => ({
+  factory: (require) => ({
     hooks: {
       /** @type {(project: Project, options: InstallOptions) => void} */
       afterAllInstalled(project, options) {
@@ -25,6 +25,7 @@ module.exports = {
           return;
         }
 
+        const { npath } = require("@yarnpkg/fslib");
         const fs = require("node:fs");
         const path = require("node:path");
 
@@ -33,18 +34,10 @@ module.exports = {
           "node_modules/react-native-macos/third-party-podspecs/boost.podspec",
         ];
 
-        /**
-         * @param {string} p
-         * @returns {string}
-         */
-        function normalize(p) {
-          // On Windows, paths are prefixed with `/`
-          return p.replace(/^[/\\]([^/\\]+:[/\\])/, "$1");
-        }
-
         for (const ws of project.workspaces) {
           for (const boostPodspec of boostPodspecs) {
-            const podspecPath = path.join(normalize(ws.cwd), boostPodspec);
+            const workspaceDir = npath.fromPortablePath(ws.cwd);
+            const podspecPath = path.join(workspaceDir, boostPodspec);
             if (!fs.existsSync(podspecPath)) {
               continue;
             }

--- a/.yarn/plugins/link-project.cjs
+++ b/.yarn/plugins/link-project.cjs
@@ -5,11 +5,11 @@
  * @typedef {{ configuration: Configuration; cwd: string; workspaces: Workspace[]; }} Project
  * @typedef {{ mode?: "skip-build" | "update-lockfile"; }} InstallOptions
  *
- * @type {{ name: string; factory: (require: NodeRequire) => unknown; }}
+ * @type {{ name: string; factory: (require: NodeJS.Require) => unknown; }}
  */
 module.exports = {
   name: "plugin-link-project",
-  factory: (_require) => ({
+  factory: (require) => ({
     hooks: {
       /** @type {(project: Project, options: InstallOptions) => void} */
       afterAllInstalled(project, options) {
@@ -25,23 +25,15 @@ module.exports = {
           return;
         }
 
+        const { npath } = require("@yarnpkg/fslib");
         const fs = require("node:fs");
         const path = require("node:path");
-
-        /**
-         * @param {string} p
-         * @returns {string}
-         */
-        function normalize(p) {
-          // On Windows, paths are prefixed with `/`
-          return p.replace(/^[/\\]([^/\\]+:[/\\])/, "$1");
-        }
 
         const noop = () => null;
         const mkdirOptions = { recursive: true, mode: 0o755 };
         const rmOptions = { force: true, maxRetries: 3, recursive: true };
 
-        const projectRoot = normalize(project.cwd);
+        const projectRoot = npath.fromPortablePath(project.cwd);
         const manifestPath = path.join(projectRoot, "package.json");
 
         fs.readFile(manifestPath, { encoding: "utf-8" }, (_err, manifest) => {
@@ -52,7 +44,8 @@ module.exports = {
               continue;
             }
 
-            const nodeModulesDir = path.join(normalize(ws.cwd), "node_modules");
+            const workspaceDir = npath.fromPortablePath(ws.cwd);
+            const nodeModulesDir = path.join(workspaceDir, "node_modules");
             const linkPath = path.join(nodeModulesDir, name);
 
             fs.readlink(linkPath, (err, linkString) => {

--- a/.yarn/plugins/npm-workaround.cjs
+++ b/.yarn/plugins/npm-workaround.cjs
@@ -5,11 +5,11 @@
  * @typedef {{ configuration: Configuration; cwd: string; workspaces: Workspace[]; }} Project
  * @typedef {{ mode?: "skip-build" | "update-lockfile"; }} InstallOptions
  *
- * @type {{ name: string; factory: (require: NodeRequire) => unknown; }}
+ * @type {{ name: string; factory: (require: NodeJS.Require) => unknown; }}
  */
 module.exports = {
   name: "plugin-npm-workaround",
-  factory: (_require) => ({
+  factory: (require) => ({
     hooks: {
       /** @type {(project: Project, options: InstallOptions) => void} */
       afterAllInstalled(project, options) {
@@ -25,6 +25,7 @@ module.exports = {
           return;
         }
 
+        const { npath } = require("@yarnpkg/fslib");
         const fs = require("node:fs");
         const path = require("node:path");
 
@@ -32,18 +33,10 @@ module.exports = {
           "node_modules/@react-native-community/cli/build/tools/npm.js",
         ];
 
-        /**
-         * @param {string} p
-         * @returns {string}
-         */
-        function normalize(p) {
-          // On Windows, paths are prefixed with `/`
-          return p.replace(/^[/\\]([^/\\]+:[/\\])/, "$1");
-        }
-
         for (const ws of project.workspaces) {
           for (const file of filesToPatch) {
-            const jsPath = path.join(normalize(ws.cwd), file);
+            const workspaceDir = npath.fromPortablePath(ws.cwd);
+            const jsPath = path.join(workspaceDir, file);
             if (!fs.existsSync(jsPath)) {
               continue;
             }

--- a/.yarn/plugins/undo-bin-sorting.cjs
+++ b/.yarn/plugins/undo-bin-sorting.cjs
@@ -1,0 +1,71 @@
+// @ts-check
+/**
+ * @typedef {{ values: Map<string, unknown>; }} Configuration
+ * @typedef {{ cwd: string; }} Workspace
+ * @typedef {{ configuration: Configuration; cwd: string; workspaces: Workspace[]; }} Project
+ * @typedef {{ mode?: "skip-build" | "update-lockfile"; }} InstallOptions
+ */
+
+/**
+ * Yarn always sorts `package.json` during install. This is problematic because
+ * we need the `bin` field to be in a specific order to workaround an issue with
+ * `bunx` always picking the first binary.
+ *
+ * For more context, see:
+ *
+ *   - https://github.com/microsoft/react-native-test-app/issues/2417
+ *   - https://github.com/yarnpkg/berry/issues/6184
+ *
+ * @type {{ name: string; factory: (require: NodeJS.Require) => unknown; }}
+ */
+module.exports = {
+  name: "plugin-undo-bin-sorting",
+  factory: (require) => {
+    const { npath } = require("@yarnpkg/fslib");
+    const fs = require("node:fs");
+
+    const asText = /** @type {const} */ ({ encoding: "utf-8" });
+
+    let manifestPath = "";
+    let orig_rawManifest = "";
+    return {
+      hooks: {
+        /** @type {(project: Project) => void} */
+        validateProject(project) {
+          const pp = npath.join(project.cwd, "package.json");
+          manifestPath = npath.fromPortablePath(pp);
+          orig_rawManifest = fs.readFileSync(manifestPath, asText);
+        },
+        /** @type {(project: Project, options: InstallOptions) => void} */
+        afterAllInstalled() {
+          const rawManifest = fs.readFileSync(manifestPath, asText);
+          if (rawManifest === orig_rawManifest) {
+            return;
+          }
+
+          const manifest = JSON.parse(rawManifest);
+          const bin = Object.keys(manifest.bin);
+
+          const orig_manifest = JSON.parse(orig_rawManifest);
+          const orig_bin = Object.keys(orig_manifest.bin);
+
+          const length = bin.length;
+          if (length !== orig_bin.length) {
+            throw new Error("Did Yarn add something to the 'bin' field?");
+          }
+
+          for (let i = 0; i < length; ++i) {
+            if (bin[i] !== orig_bin[i]) {
+              manifest.bin = orig_manifest.bin;
+              const fd = fs.openSync(manifestPath, "w", 0o644);
+              fs.writeSync(fd, JSON.stringify(manifest, undefined, 2));
+              fs.writeSync(fd, "\n");
+              fs.closeSync(fd);
+              break;
+            }
+          }
+        },
+      },
+    };
+  },
+};

--- a/.yarn/plugins/undo-bin-sorting.cjs
+++ b/.yarn/plugins/undo-bin-sorting.cjs
@@ -23,6 +23,7 @@ module.exports = {
   factory: (require) => {
     const { npath } = require("@yarnpkg/fslib");
     const fs = require("node:fs");
+    const path = require("node:path");
 
     const asText = /** @type {const} */ ({ encoding: "utf-8" });
 
@@ -32,8 +33,8 @@ module.exports = {
       hooks: {
         /** @type {(project: Project) => void} */
         validateProject(project) {
-          const pp = npath.join(project.cwd, "package.json");
-          manifestPath = npath.fromPortablePath(pp);
+          const projectRoot = npath.fromPortablePath(project.cwd);
+          manifestPath = path.join(projectRoot, "package.json");
           orig_rawManifest = fs.readFileSync(manifestPath, asText);
         },
         /** @type {(project: Project, options: InstallOptions) => void} */

--- a/.yarnrc.yml
+++ b/.yarnrc.yml
@@ -25,5 +25,6 @@ plugins:
   - path: .yarn/plugins/link-project.cjs
   - path: .yarn/plugins/boost-workaround.cjs
   - path: .yarn/plugins/npm-workaround.cjs
+  - path: .yarn/plugins/undo-bin-sorting.cjs
 tsEnableAutoTypes: false
 yarnPath: .yarn/releases/yarn-4.6.0.cjs

--- a/package.json
+++ b/package.json
@@ -56,9 +56,9 @@
   ],
   "main": "scripts/configure-projects.js",
   "bin": {
-    "configure-test-app": "scripts/configure.mjs",
     "init": "scripts/init.mjs",
     "init-test-app": "scripts/init.mjs",
+    "configure-test-app": "scripts/configure.mjs",
     "install-windows-test-app": "windows/test-app.mjs"
   },
   "repository": {

